docs/generator.go

```diff
@@ -21,43 +21,53 @@ func main() {
 	if err != nil {
 		log.Fatal(err)
 	}
+	// detect if file exists
+	_, err = os.Stat(tmpl)
+	// create file if not exists
+	if os.IsNotExist(err) {
+		var file, err = os.Create(tmpl)
+		if err != nil {
+			log.Fatal(err)
+		}
+		file.Close()
+	}
 	f, err := os.OpenFile(tmpl, os.O_APPEND, 0666)
 	value := gjson.GetBytes(content, "properties")
 	value.ForEach(func(key, value gjson.Result) bool {
-		_, err = f.WriteString("\n# " + strings.Title(key.String()) + "\n" + "This section is of " + "{{.Properties." + strings.Title(key.String()) + ".Type}} type that tells about the basic information of the user and consists of the following sub-sections:\n") //print name of sections
+		_, err = f.WriteString("\n# " + strings.Title(key.String()) + "\nThis section is of " + "{{.Properties." + strings.Title(key.String()) + ".Type}} type that tells about the basic information of the user and consists of the following sub-sections:\n") //print name of sections
 		if err != nil {
 			log.Fatal(err)
 		}
 		println(key.String())
-		result := gjson.GetBytes(content, "properties."+key.String())
+		result := gjson.GetBytes(content, "properties."+ key.String())
 		result.ForEach(func(key1, value gjson.Result) bool {
-			_, err = f.WriteString("##    " + key1.String() + "" + "\n") //print name of sub-sections
+			_, err = f.WriteString("##" + spaceGen(3) + key1.String() + "\n") //print name of sub-sections
 			if err != nil {
```

Comment:
Same as above, maybe this could be replaced by a scoped declaration in the if statement ? Not sure again, what seems best from a readability standpoint